framework/src/headers/ip/v6/mod.rs

```diff
@@ -188,25 +188,41 @@ impl Ipv6Header {
     #[inline]
     pub fn flow(&self) -> Option<FlowV6> {
         if let Some(next_hdr) = self.next_header() {
-            let src_ip = self.src();
-            let dst_ip = self.dst();
-            if self.payload_size(0) >= 4 {
-                unsafe {
-                    let self_as_u8 = (self as *const Ipv6Header) as *const u8;
-                    let port_as_u8 = self_as_u8.offset(self.offset() as isize);
+            unsafe {
+                let self_as_u8 = (self as *const Ipv6Header) as *const u8;
+
+                let mut next_hdr = next_hdr as u8;
+                let mut payload_offset = 0; //track to make sure we don't go beyond v6 payload size
+
+                // seek ahead until we find either a tcp or udp header, or exhaust the payload
+                while next_hdr != TCP_NXT_HDR && next_hdr != UDP_NXT_HDR && self.payload_size(0) > payload_offset {
+                    // start at beginning of the current ext header, the first byte is the next header,
+                    // the second byte is the header length in 8-octet unit excluding the first 8 octets
+                    let seek_to = self_as_u8.offset((self.offset() + payload_offset) as isize);
+                    next_hdr = seek_to.read();
+                    // to seek past the current ext header, we advance the offset by its size
+                    // which is (hdr_ext_len * 8 + 8) in bytes
+                    payload_offset += seek_to.offset(1).read() as usize * 8 + 8;
```

Comment:
so this is how to skip an ipv6 ext header, right? i wonder if we should be validating that these are ext headers as we parse through? otherwise this payload_offset math probably doesn't work (like if there are no ext headers and it's just ATP or some madness). that'd _probably_ just eventually run out of packet to process, but there's potentially a lot of random bits in between here and there. seems like we should be able to just check that it's one of the next_header codes for ext headers. or just the routing header, if that's all we currently support

Reply:
it also can skip right to tcp/udp if we have a regular v6. We do have a partially-filled out enum for the ext types that's available. Don't want to bound the function for regular v6 needs, but using the enum could help, maybe?

Reply:
I mean, you could just break out of this loop if you run into an unsupported `next_header`, yeah? or add `is_supported(next_hdr)` to the while conditional. and then in the next block check for TCP/UDP again.

a big TODO for validation/cleanup is also fine -- but if fixing it is fairly simple, i'd rather avoid obvious undefined behavior.

Reply:
yeah, if we see something unsupported, it's probably a bad sign anyhow.

Reply:
a tiny little TODO added.